### PR TITLE
Add feature for periodic observation of merging of observations into one architecture

### PR DESCRIPTION
**Pull request by schmerl**

### CODE REVIEW

File: src/rosdiscover/cli.py [hunk 1]
```diff
@@ -256,6 +282,9 @@ def main(args: t.Optional[t.Sequence[str]] = None) -> None:
                               formatter_class=MultiLineFormatter)
     p.add_argument('--acme', action='store_true', help='Generate an Acme file instead of the YAML')
     p.add_argument('--output', type=str, help='What file to output')
+    p.add_argument('--repeat', action='store_true', help='Indicate that observe should be repeated until Ctrl-C')
```
Comment (ChrisTimperley) [inline src/rosdiscover/cli.py:285]: is `continuous` a better name?

File: src/rosdiscover/cli.py [hunk 2]
```diff
@@ -256,6 +282,9 @@ def main(args: t.Optional[t.Sequence[str]] = None) -> None:
...
+    p.add_argument('--num-observations', type=int, help='The number of times to observe')
+    p.add_argument('--period', type=int, help='The number of seconds to wait in between observations')
```
Comment (ChrisTimperley) [inline src/rosdiscover/cli.py:287]: imo, `interval` is a slightly better name than `period`

File: src/rosdiscover/interpreter/summary.py [hunk 3]
```diff
@@ -142,3 +207,16 @@ def unresolved(self) -> Iterator[NodeSummary]:
         for n in self._node_to_summary.values():
             if n.placeholder:
                 yield n
+
+    @classmethod
+    def merge(cls, sum1: 'SystemSummary', sum2: 'SystemSummary') -> 'SystemSummary':
```
Comment (ChrisTimperley) [inline src/rosdiscover/interpreter/summary.py:212]: suggest `lhs` and `rhs` over `sum1` and `sum2`

File: src/rosdiscover/interpreter/summary.py [hunk 4]
```diff
@@ -142,3 +207,16 @@ def unresolved(self) -> Iterator[NodeSummary]:
...
+        node_summaries = {}
```
Comment (ChrisTimperley) [inline src/rosdiscover/interpreter/summary.py:213]: type annotation is probably needed here? If not, `t.Dict[t.Any]` will be assumed, I think? (Could be wrong.)

File: src/rosdiscover/cli.py [hunk 5]
```diff
@@ -114,8 +115,33 @@ def _observe(args) -> SystemSummary:
...
+            go = False
+
+    return summary
+
+
 def observe(args) -> None:
-    summary = _observe(args)
+    if args.repeat:
+        if not args.period:
+            logger.error('--repeat specified but no period given')
```
Comment (ChrisTimperley) [inline src/rosdiscover/cli.py:141]: should this crash / raise an exception?

**Comment by schmerl** [comment]
All the above suggestions have been made.

**Pull request merged**